### PR TITLE
2808 - es6 conversion

### PR DESCRIPTION
**Pull request by eleanor-heath**
https://github.com/adaptlearning/adapt_framework/issues/2808

### CODE REVIEW

File: js/adapt-extension.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
-define(function(require) {
-
-    var Adapt = require('coreJS/adapt');
-    var Backbone = require('backbone');
-    var ExtensionView = require('extensions/adapt-extension/js/adapt-extensionView');
+import Adapt from 'coreJS/adapt';
```
Comment (cahirodoherty-learningpool) [inline js/adapt-extension.js:1]: Minor: 'core/js/adapt' ?

File: js/adapt-extensionView.js [hunk 2]
```diff
@@ -1,52 +1,49 @@
...
-            // This way postRender can manipulate this view after it has
-            // been rendered
-            _.defer(_.bind(function() {
-                this.postRender();
-            }, this));
-
-            // Return this so we can change the render method
-            // this.render().$el.addClass('explode');
-            return this;
+import Adapt from 'coreJS/adapt';
```
Comment (cahirodoherty-learningpool) [inline js/adapt-extensionView.js:1]: Minor: 'core/js/adapt'

Comment (cahirodoherty-learningpool) [inline js/adapt-extensionView.js]:
```suggestion
import Adapt from 'core/js/adapt';
```

File: js/adapt-extension.js (hunk 1 above)
Comment (cahirodoherty-learningpool) [inline js/adapt-extension.js]:
```suggestion
import Adapt from 'core/js/adapt';
```

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by ethan-lp — APPROVED** [review]

File: js/adapt-extension.js [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+import Adapt from 'core/js/adapt';
+import Backbone from 'backbone';
+import ExtensionView from 'extensions/adapt-extension/js/adapt-extensionView';
 
+export default class Extension extends Backbone.Controller {
+  initialize() {
     // Listen to when the data is all loaded
-    Adapt.on('app:dataReady', function() {
-        console.log('Plugin has loaded and data is ready');
+    Adapt.listenTo('app:dataReady', function() {
```
Comment (oliverfoster) [inline js/adapt-extension.js]:
```js
  this.listenTo(Adapt, 'app:dataReady', this.onDataReady);
}

onDataReady() {
  console.log('Plugin has loaded and data is ready');
}
```

File: js/adapt-extension.js [hunk 4]
```diff
@@ -1,12 +1,12 @@
-define(function(require) {
-
-    var Adapt = require('coreJS/adapt');
-    var Backbone = require('backbone');
-    var ExtensionView = require('extensions/adapt-extension/js/adapt-extensionView');
+import Adapt from 'core/js/adapt';
+import Backbone from 'backbone';
+import ExtensionView from 'extensions/adapt-extension/js/adapt-extensionView';
```
Comment (oliverfoster) [inline js/adapt-extension.js]: call the view class file: `ExtensionView.js`

**Review by oliverfoster — CHANGES_REQUESTED** [review]

File: js/adapt-extension.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import Adapt from 'core/js/adapt';
 import Backbone from 'backbone';
-import ExtensionView from 'extensions/adapt-extension/js/adapt-extensionView';
+import ExtensionView from 'extensions/adapt-extension/js/ExtensionView.js';
```
Comment (oliverfoster) [inline js/adapt-extension.js]:
```suggestion
import ExtensionView from 'extensions/adapt-extension/js/ExtensionView';
```

don't need the `.js` here.

File: js/adapt-extension.js (hunk 3 above)
Comment (oliverfoster) [reply js/adapt-extension.js]: Missed this bit:

```js
  this.listenTo(Adapt, 'app:dataReady', this.onDataReady);
```

**Review by oliverfoster — APPROVED** [review]

**Pull request merged**